applications/acdc/src/acdc_queue_manager.erl

```diff
@@ -947,4 +958,47 @@ update_properties(QueueJObj, State) ->
     State#state{
       enter_when_empty=kz_json:is_true(<<"enter_when_empty">>, QueueJObj, 'true')
                ,moh=kz_json:get_ne_value(<<"moh">>, QueueJObj)
+               ,announcements_config=announcements_config(QueueJObj)
      }.
+
+-spec announcements_config(kz_json:object()) -> kz_proplist().
+announcements_config(Config) ->
+    kz_json:recursive_to_proplist(
+      kz_json:get_json_value(<<"announcements">>, Config, kz_json:new())).
+
+-spec cancel_position_announcements(kapps_call:call() | 'false', map()) ->
+                                           map().
+cancel_position_announcements('false', Pids) -> Pids;
+cancel_position_announcements(Call, Pids) ->
+    CallId = kapps_call:call_id(Call),
+    case maps:is_key(CallId, Pids) of
+        'true' ->
+            lager:debug("cancelling announcements for ~s", [CallId]),
+            {Pid, Pids1} = maps:take(CallId, Pids),
+            acdc_announcements_sup:stop_announcements(Pid),
+
+            %% Attempt to skip remaining announcement media, but don't flush hangups
+            NoopId = kz_datamgr:get_uuid(),
+            Command = [{<<"Application-Name">>, <<"noop">>}
+                      ,{<<"Msg-ID">>, NoopId}
+                      ,{<<"Insert-At">>, <<"now">>}
+                      ,{<<"Filter-Applications">>, [<<"play">>, <<"say">>, <<"play">>]}
+                      ],
+            kapps_call_command:send_command(Command, Call),
+            Pids1;
+        'false' ->
+            lager:debug("did not have the announcements for call ~s", [CallId]),
+            Pids
+    end.
+
+-spec maybe_remove_queue_member(api_binary(), mgr_state()) -> mgr_state().
+maybe_remove_queue_member(CallId, #state{current_member_calls=CurrentCalls
```

Comment:
A small nitpick, but "maybe_*" implies that it may or may not do the action, this will always remove the member.